data_model/src/predicate.rs

```diff
@@ -881,3 +1067,141 @@ pub mod value {
+                hexadecuplet_0.into(),
+                hexadecuplet_1.into(),
+                hexadecuplet_2.into(),
+                hexadecuplet_3.into(),
+                hexadecuplet_4.into(),
+                hexadecuplet_5.into(),
+                hexadecuplet_6.into(),
+                hexadecuplet_7.into(),
+            ])
+        }
+    }
+
+    #[cfg(test)]
+    mod test {
+        #![allow(clippy::restriction)]
+        use super::*;
+
+        #[test]
+        fn ipv4_filter_example() {
+            {
+                let pred = Ipv4Predicate::new(127, 0, 0, (0, 10));
+                println!("{pred:?}");
+                assert!(pred.applies(Ipv4Addr([127, 0, 0, 1])));
+                assert!(pred.applies(Ipv4Addr([127, 0, 0, 3])));
+                assert!(pred.applies(Ipv4Addr([127, 0, 0, 4])));
+                assert!(pred.applies(Ipv4Addr([127, 0, 0, 10])));
+                assert!(!pred.applies(Ipv4Addr([127, 0, 0, 11])));
+                assert!(!pred.applies(Ipv4Addr([125, 0, 0, 1])));
+                assert!(!pred.applies(Ipv4Addr([128, 0, 0, 1])));
+                assert!(!pred.applies(Ipv4Addr([127, 1, 0, 1])));
+                assert!(!pred.applies(Ipv4Addr([127, 0, 1, 1])));
+            }
+
+            {
+                let pred = Ipv4Predicate::new(Mask::starting(0), 0, 0, (0, 10));
+                println!("{pred:?}");
+                assert!(pred.applies(Ipv4Addr([0, 0, 0, 1])));
+                assert!(pred.applies(Ipv4Addr([255, 0, 0, 1])));
+                assert!(pred.applies(Ipv4Addr([127, 0, 0, 4])));
+                assert!(pred.applies(Ipv4Addr([127, 0, 0, 10])));
+                assert!(pred.applies(Ipv4Addr([128, 0, 0, 1])));
+                assert!(pred.applies(Ipv4Addr([128, 0, 0, 1])));
+                assert!(!pred.applies(Ipv4Addr([127, 0, 0, 11])));
+                assert!(pred.applies(Ipv4Addr([126, 0, 0, 1])));
+                assert!(pred.applies(Ipv4Addr([128, 0, 0, 1])));
+                assert!(!pred.applies(Ipv4Addr([127, 1, 0, 1])));
+                assert!(!pred.applies(Ipv4Addr([127, 0, 1, 1])));
+            }
+        }
+
+        #[test]
+        fn ipv6_filter_example() {
+            let pred = Ipv6Predicate::new(12700, 0, 0, (0, 10), 0, 0, 0, 0);
+            println!("{pred:?}");
+            assert!(pred.applies(Ipv6Addr([12700, 0, 0, 1, 0, 0, 0, 0])));
+            assert!(pred.applies(Ipv6Addr([12700, 0, 0, 3, 0, 0, 0, 0])));
+            assert!(pred.applies(Ipv6Addr([12700, 0, 0, 4, 0, 0, 0, 0])));
+            assert!(pred.applies(Ipv6Addr([12700, 0, 0, 10, 0, 0, 0, 0])));
+            assert!(!pred.applies(Ipv6Addr([12700, 0, 0, 11, 0, 0, 0, 0])));
+            assert!(!pred.applies(Ipv6Addr([12500, 0, 0, 1, 0, 0, 0, 0])));
+            assert!(!pred.applies(Ipv6Addr([12800, 0, 0, 1, 0, 0, 0, 0])));
+            assert!(!pred.applies(Ipv6Addr([12700, 1, 0, 1, 0, 0, 0, 0])));
+            assert!(!pred.applies(Ipv6Addr([12700, 0, 1, 1, 0, 0, 0, 0])));
```

Comment:
Edge cases would be enough:

```suggestion
            let pred = Ipv6Predicate::new(12700, 0, 0, (0, 10), 0, 0, 0, 0);
            println!("{pred:?}");
            assert!(pred.applies(Ipv6Addr([12700, 0, 0, 0, 0, 0, 0, 0])));
            assert!(pred.applies(Ipv6Addr([12700, 0, 0, 10, 0, 0, 0, 0])));
            assert!(!pred.applies(Ipv6Addr([12700, 0, 0, 11, 0, 0, 0, 0])));
            assert!(!pred.applies(Ipv6Addr([12700, 1, 0, 1, 0, 0, 0, 0])));
```

Reply:
This is necessary for full coverage.